packages/permissionless/errors/account.ts

```diff
@@ -0,0 +1,306 @@
+                "• If you are not looking to use time-based signatures, verify that the validAfter and validUntil fields are set to 0.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountRevertedOrOutOfGasDuringValidation extends BaseError {
+    static message = /aa23/
+    override name = "SmartAccountRevertedOrOutOfGasDuringValidation"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `The smart account ${sender} reverted or ran out of gas during the validation of the user operation.`,
+                "",
+                "Possible solutions:",
+                "• Verify that the verificationGasLimit is high enough to cover the validateUserOp function's gas costs.",
+                "• Make sure validateUserOp returns uint(1) for invalid signatures, and MUST NOT REVERT when the signature is invalid",
+                "• If you are not using a paymaster, verify that the sender address has enough native tokens to cover the required pre fund. Consider leveraging functions like getRequiredPrefund.",
+                "• Verify that the validateUserOp function is implemented with the correct logic, and that the user operation is supposed to be valid.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountInvalidSignature extends BaseError {
+    static message = /aa24/
+    override name = "SmartAccountInvalidSignature"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `The smart account ${sender} signature is invalid.`,
+                "",
+                "Possible solutions:",
+                "• Verify that the user operation was correctly signed, and that the signature was correctly encoded in the signature field of the user operation.",
+                "• Most smart account implementations sign over the userOpHash. Make sure that the userOpHash is correctly computed. Consider leveraging functions like getUserOperationHash.",
+                "• Make sure you have selected the correct chainId and entryPointAddress when computing the userOpHash.",
+                "• Make sure the smart account signature verification function is correctly implemented.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountNonceInvalid extends BaseError {
```

Comment:
SmartAccountNonceInvalid -> NonceInvalid?

Reply:
Can do that, but similarly thought everything has SmartAccount in front of it